Lib/test/test_asyncio/test_base_events.py

```diff
@@ -926,6 +926,54 @@ def test_run_forever_pre_stopped(self):
         self.loop.run_forever()
         self.loop._selector.select.assert_called_once_with(0)
 
+    async def leave_unfinalized_asyncgen(self):
+        # The following should create an async generator, iterate
+        # it partially, and leave it to be garbage collected
+        # in the future.
```

Comment:
This coroutine does what it says now, in cpython. It's probably dependent on how precisely reference counting and garbage collection works, and may stop working in future versions or in other implementations. Is it OK to rely on this behavior in the test below?

Reply:
I think it's OK, just document this fact.